### ISSUE TITLE
Bug with API calls, async loading, persistence, etc.

### ISSUE THREAD

**Issue by pwalsh** [okfn/spend-publishing-dashboard#19]
The codebase has some bugs with loading data that I haven't yet investigated.

When changing routes, I was experiencing weird behaviour, like not having data available.

Currently, the app is rather simple:
- React for the view layer: http://facebook.github.io/react/
- Flux datastore for the model(s): http://facebook.github.io/flux/

Here is the flow:
- When app initializes, we call data here: https://github.com/okfn/spend-publishing-dashboard/blob/master/dashboard/src/scripts/app.js#L17
- When the data arrives from the API, an action is created: https://github.com/okfn/spend-publishing-dashboard/blob/master/dashboard/src/scripts/actions/ServerActionCreators.js#L9
- This action updates the relevant store, for example: https://github.com/okfn/spend-publishing-dashboard/blob/master/dashboard/src/scripts/stores/publisherStore.js#L55
- Which then emits an event: https://github.com/okfn/spend-publishing-dashboard/blob/master/dashboard/src/scripts/stores/publisherStore.js#L57
- So that the UI can re-render with data: https://github.com/okfn/spend-publishing-dashboard/blob/master/dashboard/src/scripts/components/pages/Mixins.react.js#L14

The first load (home page) seems to work, but changing routes has unpredictable behaviour.

That is where I left it. Need to look in and see what is happening. There is likely a problem with emitting events, or persisting data over routes.

**Comment by hdurand** [comment]
I think the pull request #22 fixes it by fixing a CSV parsing error.
To be confirmed.

**Comment by hdurand** [comment]
Hi @pwalsh 
- Does the following describes this bug:
  sometimes, in a table, when clicking a publisher or source id link, the url changes but nothing else happens?
  If it does, then it has been fixed by pull request #22.
- Otherwise, what is it exactly?

**Comment by pwalsh** [comment]
Hi. No, that is not the behaviour I saw. I saw:
- On changing route by clicking a publisher link (for example), the source table for that publisher would return empty (as if the data was not loaded)
- Directly loading a route that is not the home (eg: try to refresh in the browser after you have navigated to a publisher page) would fail

I understood from when we talked last week that these problems were solved by the change you made in #22, but I have  ot run the code myself to confirm this.
